### ISSUE TITLE
[Bug]: Spaced symbols should not be spaced when in limits etc

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#26174]
### Bug summary

As seen, LaTeX does not add a space around the arrows (or =/+-sign) when in subscripts, but mathtext does.

Note that there is no space around `\to` but  in mathtext,around `\rightarrow` in mathtext, while for LaTeX they look exactly the same. (This is because `\to` is not listed in `_mathtext.Parser._arrow_symbols`, but it should be.)

### Code for reproduction

```python

import matplotlib.pyplot as plt

strings = [r'$\limsup_{x\to\infty}$',
           r'$\limsup_{x\rightarrow\infty}$',
           r'$x \to \infty$',
           r'$x \rightarrow \infty$',
           r'$\limsup_{x=a+b}$',
           r'$x=a+b$',
           r'$A^{a \times b}_{c \Uparrow d}$']

for i, s in enumerate(strings):
    plt.text(0, i*0.15 + 0.05, s, fontsize=20)
    plt.text(0.5, i*0.15 + 0.05, f"${s}$", usetex=True, fontsize=20)

plt.axis(False)
plt.savefig("spacing.png", dpi=200)
```

### Actual outcome

![image](https://github.com/matplotlib/matplotlib/assets/8114497/b3b3b4fc-910e-4314-be45-0e3ad8b7b164)

### Expected outcome

No spaces around "spaced operators" in subscript.

### Additional information

@devRD is this something you know how to fix?

Related to the failing test in https://github.com/matplotlib/matplotlib/pull/26173

### Matplotlib Version

Latest main

### Installation

None

**Comment by devRD** [comment]
All the equations get parsed through the `def symbol:` function, so in mathtext we have the following code to add spacing around the operators. Currently, we have a check for binary operators to neglect spacing at the start.

https://github.com/matplotlib/matplotlib/blob/623da30baee382905c3f06aa014d7d9841d02684/lib/matplotlib/_mathtext.py#L2128-L2131

 For this issue, we could add a check for limits so that the symbols are not spaced by mathtext?

**Comment by anntzer** [comment]
(Side point re: the original report: the fair comparison with latex would use `$$...$$` in latex as we're effectively always in displaymath mode in mathtext, but even there the spacing gets suppressed.)

**Comment by oscargus** [comment]
The same issue happens in sub- and superscripts as well. I checked the code a bit and maybe the easiest way is to remove any space from sub- and superscripts (and over/under)?

(Re: side point: I was too lazy to sort that out, but have updated the example, at least if you wait 30 seconds after this comment.)

**Comment by oscargus** [comment]
Or maybe one can use `self._in_subscript_or_superscript` to determine if spaces should be added? (Have not studied the code enough to realize if that may work.)

I guess blindly removing spaces may also remove those added by e.g. `\quad`...

**Comment by oscargus** [comment]
@devRD, I fixed it in #26173 as it turned out to be straightforward (do not add space if `self._in_subscript_or_superscript`).

**Comment by oscargus** [comment]
Well, actually not fully. It works for the overunder case, but not for sub- and superscripts...

**Comment by devRD** [comment]
The `self._in_subscript_or_superscript` check does work for the over-under cases!

I think the `self._in_subscript_or_superscript` flag determines if we need to add spacing to the nucleus, and it is not checking the subscripts and superscripts so for the string like `r'$A^{a \times b}_{c \Uparrow d}$` in the example the flag's value is `False`

**Comment by oscargus** [comment]
Agreed. I've tried to insert a separate flag in `subsuper`, but that didn't work...
